fdbserver/DataDistribution.actor.cpp

```diff
@@ -2246,7 +2258,21 @@ ACTOR Future<Void> makeAuditProgressOnServer(Reference<DataDistributor> self,
 					audit->foundError = true;
 				} else {
 					ASSERT(phase == AuditPhase::Invalid);
-					AuditStorageRequest req(audit->coreState.id, auditState.range, audit->coreState.getType());
+					// Set doAuditOnStorageServer
+					if (self->remainingBudgetForAuditTasks.contains(auditType)) {
+						ASSERT(self->remainingBudgetForAuditTasks[auditType].get() >= 0);
+						while (self->remainingBudgetForAuditTasks[auditType].get() == 0) {
+							wait(self->remainingBudgetForAuditTasks[auditType].onChange());
+							ASSERT(self->remainingBudgetForAuditTasks[auditType].get() >= 0);
+						}
+						self->remainingBudgetForAuditTasks[auditType].set(
```

Comment:
Is it a little late to do the budget limiting here, since we already created # of SS ACTORs?

Reply:
No. We create `doAuditOnStorageServer` as well as SS actors after the budget checking. So, if no budge, no actor of `doAuditOnStorageServer` and corresponding SS actor will be created until the budge updated when some else doAuditOnStorageServer actor completes.